nighthawk/include/nighthawk/common/worker.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+#include "nighthawk/common/statistic.h"
+
+namespace Nighthawk {
+
+/**
+ * Interface for a threaded worker.
+ */
+class Worker {
+public:
+  virtual ~Worker() {}
+
+  /**
+   * Start the worker thread.
+   */
+  virtual void start() PURE;
+
+  /**
+   * Wait for the worker thread to complete its work.
+   */
+  virtual void waitForCompletion() PURE;
+
+  /**
+   * Perform the actual work on the associated thread initiated by start().
+   */
+  virtual void work() PURE;
```

Comment:
Why is this in the interface? Isn't it only a `WorkerImpl` detail?

Reply:
Pinged you on slack to discuss